projects/core/src/event/tms.event.ts

```diff
@@ -0,0 +1,26 @@
+import { Observable } from 'rxjs';
+import { createFrom } from '../util/create-from';
+import { EventService } from './event.service';
+
+export class TmsEvent {
+  static type = 'TmsEvent';
+  event: string;
+  payload: any;
+}
+
+export abstract class AbstractTmsEventCollector {
+  protected abstract events$: Observable<TmsEvent>[];
```

Comment:
Thinking about better name here... maybe `sources`? Also, it's an array, not an observable, so we should not use `$` here.